Documentation/ExtensionArchitecture/Extbase/Reference/_FileUpload/_ApiUpload.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Controller;
+
+use MyVendor\MyExtension\Domain\Model\Blog;
+use MyVendor\MyExtension\Domain\Repository\BlogRepository;
+use TYPO3\CMS\Core\Resource\DuplicationBehavior;
+use TYPO3\CMS\Core\Resource\ResourceFactory;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\StringUtility;
+use TYPO3\CMS\Extbase\Domain\Model\FileReference;
+use TYPO3\CMS\Extbase\Mvc\Controller\ActionController;
+
+class BlogController extends ActionController
+{
+    public function __construct(
+        protected ResourceFactory $resourceFactory,
+        protected BlogRepository $blogRepository,
+    ) {}
+
+    public function attachFileUpload(Blog $blog): void
+    {
+        $falIdentifier = '1:/your_storage';
+        $yourFile = '/path/to/uploaded/file.jpg';
+
+        // Attach the file to the wanted storage
+        $falFolder = $this->resourceFactory->retrieveFileOrFolderObject($falIdentifier);
```

Comment:
T>here is no `addFile` method in `File` object. I would prefer to check result with `instanceof` first

Reply:
But the falIdentifier here is a FolderObject one. For easier readability I would like to not add complexity to the example (that one shouldn't use in v13 anyways) by adding more checks here. This is more a "proof of concept" example... would that be okay with you? Further code improvement could be done in a seperate followup.